docs/proposals/volume-provisioning.md

```diff
@@ -101,6 +98,12 @@ We propose that:
 
 ### Controller workflow for provisioning volumes
 
+0. Kubernetes administator can configure name of a default StorageClass. This
+   StorageClass instance is then used when user requests a dynamically
+   provisioned volume, but does not specify a StorageClass. In other words,
+   `claim.Spec.Class == ""`
```

Comment:
Should this potentially be expanded to something like `Params`?

Assume that StorageClass is the correct place to configure whether or not a volume is encrypted.  We need a way to pass an encryption key to the provisioner.

A new `SecretType` can be added to manage these keys for the user.  The PVC would therefore need both `claim.Spec.Class` and `claim.Spec.Secret` as params to the provisioner.

All GCE volumes are encrypted by default, so an empty Secret means GCE creates its own keypair.  AWS is *optionally* encrypted, so an empty key might mean "no encryption" or it might mean "generate a keypair".